packages/actor-tests/bedrock/deposit-eth.ts

```diff
@@ -0,0 +1,88 @@
+import * as fs from 'fs'
+
+import { utils, Wallet, providers, constants } from 'ethers'
+import {
+  CrossChainMessenger,
+  ETHBridgeAdapter,
+  StandardBridgeAdapter,
+} from '@eth-optimism/sdk'
+import { predeploys } from '@eth-optimism/contracts-bedrock'
+import { sleep } from '@eth-optimism/core-utils'
+
+import { actor, setupActor, run } from '../lib/convenience'
+
+interface Context {
+  wallet: Wallet
+}
+
+actor('Dev account sender', () => {
+  let l1Provider: providers.JsonRpcProvider
+
+  let l2Provider: providers.JsonRpcProvider
+
+  let wallet: Wallet
+
+  let messenger: CrossChainMessenger
+
+  let contracts: any
+
+  setupActor(async () => {
+    l1Provider = new providers.JsonRpcProvider(process.env.L1_RPC)
+    l2Provider = new providers.JsonRpcProvider(process.env.L2_RPC)
+    wallet = new Wallet(process.env.PRIVATE_KEY)
+    contracts = JSON.parse(
+      fs.readFileSync(process.env.CONTRACTS_JSON_PATH).toString('utf-8')
+    )
+    messenger = new CrossChainMessenger({
+      l1SignerOrProvider: wallet.connect(l1Provider),
+      l2SignerOrProvider: wallet.connect(l2Provider),
+      l1ChainId: (await l1Provider.getNetwork()).chainId,
+      l2ChainId: (await l2Provider.getNetwork()).chainId,
+      bridges: {
+        Standard: {
+          Adapter: StandardBridgeAdapter,
+          l1Bridge: contracts.L1StandardBridgeProxy,
+          l2Bridge: predeploys.L2StandardBridge,
+        },
+        ETH: {
+          Adapter: ETHBridgeAdapter,
+          l1Bridge: contracts.L1StandardBridgeProxy,
+          l2Bridge: predeploys.L2StandardBridge,
+        },
+      },
+      contracts: {
+        l1: {
+          AddressManager: constants.AddressZero,
+          StateCommitmentChain: constants.AddressZero,
+          CanonicalTransactionChain: constants.AddressZero,
+          BondManager: constants.AddressZero,
+          L1StandardBridge: contracts.L1StandardBridgeProxy,
+          L1CrossDomainMessenger: contracts.L1CrossDomainMessengerProxy,
+          L2OutputOracle: contracts.L2OutputOracleProxy,
+          OptimismPortal: contracts.OptimismPortalProxy,
+        },
+      },
+      bedrock: true,
+    })
+  })
+
+  run(async (b, ctx: Context, logger) => {
+    const recipient = Wallet.createRandom().connect(l2Provider)
+    logger.log(`Depositing funds to ${recipient.address}.`)
+    const depositTx = await messenger.depositETH(utils.parseEther('0.0001'), {
+      recipient: recipient.address,
+    })
+    logger.log(`Awaiting receipt for deposit tx ${depositTx.hash}.`)
+    await depositTx.wait()
+    // Temporary until this is supported in the SDK.
+    for (let i = 0; i < 60; i++) {
+      const recipBal = await recipient.getBalance()
```

Comment:
A less brittle way to do this would be this: https://github.com/ethereum-optimism/optimism/blob/09658f16b6e1e02dc9f589821b82746fa947fa50/packages/contracts-bedrock/tasks/deposits.ts#L92

Poll on the transaction hash that is computed client side

Reply:
https://github.com/ethereum-optimism/optimism/pull/3160 fixes the hanging, still need to debug why the value isn't being fully passed through

Reply:
Did you see my comments on #3160?

I'll update to do the client side poll.

Reply:
Hmm client side poll is more difficult since the Ethers types don't line up. The deposit TX utility in core utils expects an `Event` but because the deposit is initiated by the SDK i just get back `Logs`, which isn't compatible with `Event` even after inflating them using the Ethers `Interface` class.

Any objections to me updating this once the SDK is merged?